### DISCUSSION TITLE
unable to install policy X.X.X.X/X === Y.Y.Y.Y/Y out for reqid M, the same policy for reqid N exists

### DISCUSSION THREAD

**Discussion by ashirbadIndia** [strongswan/strongswan#2009] [Q&A] [answered]
I'm trying to use Strongswan to encrypt traffic between Kubernetes pods. I have multiple pods running with Strongswan installed in all of them, with the following config

```
connections {
 
   gw-gw {

      remote_addrs = 10.231.12.0/23, 10.231.6.0/23

      local {
         auth = psk
         id = sun.strongswan.org
      }
      remote {
         auth = psk
         id = sun.strongswan.org
      }
      children {
         trap-any {
            local_ts  = 10.231.12.0/23, 10.231.6.0/23
            remote_ts = 10.231.12.0/23, 10.231.6.0/23
            start_action = trap
            mode = transport
 
            updown = /usr/local/libexec/ipsec/_updown iptables
            rekey_time = 5400
            rekey_bytes = 500000000
            rekey_packets = 1000000
            esp_proposals = aes128gcm128-x25519
         }
      }
      version = 2
      mobike = no
      reauth_time = 10800
      proposals = aes128-sha256-x25519
   }
}
```

I have applied the same config on all systems.

IPSec is working correctly on almost all pod but failing for few pairs.

I checked a pod pair between whom it was failing, let them be pod A & pod B.

**A: ip 10.231.12.101**
**B: ip 10.231.12.100**

I tried to ping pod B from A, I can see ESP packets from A reaching B, but B is not able to reply back.

I can see SAs being created in A, but not in B. 

I see the following logs on B, where SA are not created. Phase 1 negotiation is succeeding, Phase 2 is failing with TS_UNACCEPTABLE

```
06[IKE] IKE_SA gw-gw[41] established between 10.231.12.100[sun.strongswan.org]...10.231.12.101[sun.strongswan.org]
06[IKE] IKE_SA gw-gw[41] state change: CONNECTING => ESTABLISHED
06[IKE] scheduling reauthentication in 9811s
06[IKE] maximum IKE_SA lifetime 10891s
06[IKE] received TS_UNACCEPTABLE notify, no CHILD_SA built
06[IKE] failed to establish CHILD_SA, keeping IKE_SA
```

& The following logs on A, unable to install policy X.X.X.X/X === Y.Y.Y.Y/Y out for reqid 3, the same policy for reqid 1 exists

```
07[IKE] scheduling reauthentication in 9807s
07[IKE] maximum IKE_SA lifetime 10887s
07[CFG] selected proposal: ESP:AES_GCM_16_128/NO_EXT_SEQ
07[CFG] unable to install policy 10.231.12.100/32 === 10.231.12.101/32 in for reqid 3, the same policy for reqid 1 exists
07[CFG] unable to install policy 10.231.12.101/32 === 10.231.12.100/32 out for reqid 3, the same policy for reqid 1 exists
07[IKE] unable to install IPsec policies (SPD) in kernel
07[IKE] failed to establish CHILD_SA, keeping IKE_SA
```

I can see SA already created in A with reqid 1,
```
src 10.231.12.101 dst 10.231.12.100
        proto esp spi 0xc674389e reqid 1 mode transport
        replay-window 0 
        aead rfc4106(gcm(aes)) 0xcc5183f0440b623149c58434debb712e0dedd2e8 128
        anti-replay context: seq 0x0, oseq 0x266d, bitmap 0x00000000
        sel src 10.231.12.101/32 dst 10.231.12.100/32 
src 10.231.12.100 dst 10.231.12.101
        proto esp spi 0xc5031c24 reqid 1 mode transport
        replay-window 32 
        aead rfc4106(gcm(aes)) 0xc37725340e3b67e3258aa6433be08df3dac22f55 128
        anti-replay context: seq 0x266a, oseq 0x0, bitmap 0xffffffff
        sel src 10.231.12.100/32 dst 10.231.12.101/32 
```

[pod A(10.231.12.101) charon logs.txt](https://github.com/strongswan/strongswan/files/13437538/pod.A.charon.logs.txt)
[pod B(10.231.12.100) charon logs.txt](https://github.com/strongswan/strongswan/files/13437541/pod.B.charon.logs.txt)

Please help me figure out where is the issue, what I'm doing wrong.

**Comment by tobiasbrunner** [comment, marked as answer]
Note that your logs are hard to compare as they don't contain any timestamps (see #196 for a useful snippet). And what's definitely not ideal is using the same identity on all hosts. Because if if you don't also configure `unique = never` that will cause a lot of this:

    destroying duplicate IKE_SA for peer 'sun.strongswan.org', received INITIAL_CONTACT

That is, hosts unilaterally destroying completely unrelated SAs without the peers getting notified. On the other hand, for other connections there might not be a notify even though the initiator doesn't have a previous SA with a specific target host.

That could cause the reqid issue as a side-effect, because as initiator, the reqid of the wide trap policy is maintained, however, as responder a new reqid for the narrowed, host-to-host traffic selectors will be allocated. So if host A initiates an SA to host B, but the latter kills that later due to a connection with some other host and subsequently initiates an SA to A itself (while not sending the INITIAL_CONTACT notify because it has an IKE_SA with the same identities and some other host), there could be a mismatch on A due to the existing and never deleted SA it initiated with B before.
